projects/vdk-plugins/vdk-impala/src/vdk/plugin/impala/impala_plugin.py

```diff
@@ -134,28 +48,149 @@ def impala_query(ctx: click.Context, query):
     )
 
 
-@hookimpl
-def vdk_command_line(root_command: click.Group):
-    root_command.add_command(impala_query)
+class ImpalaPlugin:
+    @hookimpl
+    def vdk_configure(self, config_builder: ConfigurationBuilder) -> None:
+        """
+        Here we define what configuration settings are needed for Impala with reasonable defaults
+        """
+        config_builder.add(
```

Comment:
Let's separate configuration in separate module  and use Wrapper class - similar to https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-plugins/vdk-plugin-control-cli/src/vdk/plugin/control_cli_plugin/control_service_configuration.py